datashare-api/src/main/java/org/icij/datashare/user/UserRepository.java

```diff
@@ -0,0 +1,15 @@
+package org.icij.datashare.user;
+
+import java.util.List;
+
+public interface UserRepository {
+    boolean save(User user);
+    User getUser(String userId);
+    UserPolicy get(User user, String projectId);
+    UserPolicy get(String userId, String projectId);
+    List<UserPolicy> list(User user);
+    List<UserPolicy> getAll();
+    boolean save(UserPolicy permission);
+    boolean delete(User user, String projectId);
+    User getUserWithPolicies(String userId);
```

Comment:
I find it confusing to have 2 methods to get users, one which populates the policy field, the other one that does not.

If it's for DB optimization (like we don't want to do the join and fetch policies) I think then it's a smell and we should keep user and policies separate. The user wouldn't have policies but we would have to API to retrieve the user policies given a user ID.

I find it a bit error prone to get user with and without populated policies; That means that later in the code, there will be different inconsistent behavior depending on how the user was loaded.

Reply:
So I would either:
- load the policies all the time
- or keep the user policies outside of the user object and create a dedicate method to get user policies given a user

Reply:
Also I would rename it as `getAllPolicies`